app/jobs/process_pull_request_update_job.rb

```diff
@@ -1,31 +1,57 @@
 class ProcessPullRequestUpdateJob < ApplicationJob
   queue_as :default
 
-  def perform(pr_data)
-    # Fetch and append the reviews which the pull request data does not contain
-    pr_data[:reviews] = reviews(pr_data[:repo], pr_data[:number])
+  def self.perform_later(pull_request_update)
+    return unless self.worth_queuing?(pull_request_update)
+
+    super(pull_request_update)
+  end
+
+  def self.worth_queuing?(pull_request_update)
+    %w[closed edited opened reopened labeled unlabeled].include?(pull_request_update[:action])
+  end
+
+  def perform(pull_request_update)
+    # Add the reviews to the pull request update as it does not have
+    # that information by default
+    add_reviews!(pull_request_update)
 
     Github::PullRequest::CreateOrUpdate.(
-      pr_data[:node_id],
-      number: pr_data[:number],
-      title: pr_data[:title],
-      author_username: pr_data[:author_username],
-      merged_by_username: pr_data[:merged_by_username],
-      repo: pr_data[:repo],
-      reviews: pr_data[:reviews],
-      data: pr_data
+      pull_request_update[:node_id],
+      number: pull_request_update[:number],
+      title: pull_request_update[:title],
+      author_username: pull_request_update[:author_username],
+      merged_by_username: pull_request_update[:merged_by_username],
+      repo: pull_request_update[:repo],
+      reviews: pull_request_update[:reviews],
+      state: state(pull_request_update),
+      data: pull_request_update
     )
 
-    User::ReputationToken::AwardForPullRequest.(pr_data)
+    User::ReputationToken::AwardForPullRequest.(pull_request_update) if award_reputation_tokens?(pull_request_update)
   end
 
   private
+  def add_reviews!(pull_request_update)
+    pull_request_update[:reviews] = reviews(pull_request_update[:repo], pull_request_update[:number])
+  end
+
+  def award_reputation_tokens?(pull_request_update)
+    %w[closed labeled unlabeled].include?(pull_request_update[:action])
+  end
+
+  def state(pull_request_update)
+    return :merged if !!pull_request_update[:merged]
+
+    pull_request_update[:state].downcase.to_sym
```

Comment:
This matches how the graphql sets a PR's state: `open`/`closed`/`merged`, where a pull request webhook has only `open`/`closed` and has an additional `merged` field.